.github/workflows/static-code-analysis.yml

```diff
@@ -39,3 +42,21 @@ jobs:
 
     - name: Perform CodeQL Analysis
       uses: github/codeql-action/analyze@v1
+
+  cppcheck:
+    if: github.event_name == 'pull_request_target'
```

Comment:
So cppcheck willl not be run on push?

Reply:
No, this action can only be run in PRs. As mentioned in the meeting I prefer to use a Github action rather than a custom check because it enables friendlier feedback by writing comments in the PR itself. I already checked several of these actions and they don't seem to work for PRs.

Reply:
Ok. I check and pull_request_target is not needed, plain pull_request is sufficient so you don't leak secrets to the foreign actions. You need to add:

```yaml
    permissions:
      pull-requests: write
```